pennylane/ops/qubit/hamiltonian.py

```diff
@@ -550,3 +550,73 @@ def queue(self, context=qml.QueuingContext):
 
         context.append(self, owns=tuple(self.ops))
         return self
+
+    def sparse_matrix(self, wires=None):
```

Comment:
I know this is controversial: Should we implement this as a function, rather than a method? My argument to do this as a method is that we find it! In 2 years, deep inside core, it will be much easier to do `op.sparse_matrix` than `qml.some_module.sparse_matrix`. And we may even add observables that only have this representation. Lastly, one can do things like `hasattr(obs, "sparse_matrix")` to find out if we can do the sparse expval evaluation!

Reply:
not controversial from my end! I think it makes a lot of sense, since we are passing the Hamiltonian object to devices; easier for the device to call a method on the object if they need a sparse representation, rather than importing PL.